src/node/config.ts

```diff
@@ -82,7 +85,12 @@ const resolve = (root: string, file: string) =>
 /**
  * Type config helper
  */
-export function defineConfig(config: RawConfigExports) {
+ export function defineConfig<T extends ThemeConfig = ThemeConfig>(
+  config: UserConfig<T>,
+  usingCustomTheme: true
```

Comment:
Motivation here: TypeScript cannot overload a function type inferring based on the number of generic type parameters, so I added an additional function parameters here.